### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bohdan-mykhailenko**
[DEMO LINK](https://bohdan-mykhailenko.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/components/TodoList/TodoList.tsx [hunk 1]
```diff
@@ -1 +1,17 @@
...
+
+export const TodoList: React.FC<TodoListProps> = ({ todos }) => {
+  return (
+    <section className="TodoList">
+      {todos.map(todo => (
+        <TodoInfo todo={todo} key={todo.id} />
+      ))}
+    </section>
+  );
+};
```
Comment (silchyn) [inline src/components/TodoList/TodoList.tsx:17]:
```suggestion
export const TodoList: React.FC<TodoListProps> = ({ todos }) => (
  <section className="TodoList">
    {todos.map(todo => (
      <TodoInfo todo={todo} key={todo.id} />
    ))}
  </section>
);
```

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/components/UserInfo/UserInfo.tsx [hunk 2]
```diff
@@ -1 +1,19 @@
...
+  user: User;
+}
+
+export const UserInfo: React.FC<UserInfoProps> = ({ user }) => {
+  const { name, email } = user;
+
+  return (
+    <a
+      className="UserInfo"
+      href={`mailto: ${email}`}
```
Comment (dinaminahi) [inline src/components/UserInfo/UserInfo.tsx:14]: to pass the tests 
```suggestion
      href={`mailto:${email}`}
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,141 @@
...
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setselectedUser] = useState('');
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserSelectError, setIsUserSelectError] = useState(false);
+
```
Comment (dinaminahi) [inline src/App.tsx:33]:
```suggestion
  const [isTitleValid, setIsTitleValid] = useState(false);
  const [isUserValid, setIsUserValid] = useState(false);

```

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,141 @@
...
+};
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setselectedUser] = useState('');
```
Comment (dinaminahi) [inline src/App.tsx:29]:
```suggestion
  const [selectedUser, setSelectedUser] = useState('');
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,141 @@
...
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserSelectError, setIsUserSelectError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
```
Comment (dinaminahi) [inline src/App.tsx:37]:
```suggestion
    setTitle(event.target.value);
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,141 @@
...
+            {'User: '}
+            <select
+              data-cy="userSelect"
+              value={selectedUser}
+              onChange={handleUserChange}
+            >
+              <option value="" disabled>Choose a user</option>
+
+              {usersFromServer.map(user => (
+                <option value={user.name} key={user.id}>
```
Comment (dinaminahi) [inline src/App.tsx:118]: Let's use user.id for the option value

**Pull request closed without merging**